src/database/include/database/context.hh

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <filesystem>
+#include <functional>
+#include <string_view>
+
+#include <sqlite3.h>
+
+namespace db
+{
+
+/**
+ * @brief Context for database operations.
+ * Contains the handle to sqlite3 database.
+ *
+ * @note This class can not be copied.
+ */
+class context
+{
+
+public:
+    /**
+     * @brief Opens the database and initializes the handle.
+     * The database is opened with write permissions.
+     *
+     * @throws std::runtime_error when the database could not be opened and prints the error to <tt>stderr</tt>.
+     * @param path Valid path to the sqlite3 database.
+     */
+    [[nodiscard]]
+    explicit
+    context(std::filesystem::path const& path);
+
+    ~context();
+
+    context(context const&) = delete;
+    context(context&&) = default;
```

Comment:
I think that the default move constructor and assignment operator do not exactly do what you intended.
It can be said that the `defaults` performs `std::move` on each class member during `context(context&&)` or `context& operator=(context&&)` but `pointer` has no `default` `move` operation implemented since (I think) the value of the `handle_` is going to be simply copied thus we perform `shallow copy` in a place where `deep copy` should be performed which is an error in my opinion.